general-concepts/dependencies/text.xml

```diff
@@ -850,5 +850,309 @@ USE flags. This would then break building your ebuild.
+</p>
+
+<codesample lang="ebuild">
+# Good:
+RDEPEND="
+	|| (
+		( sys-libs/db:5 tools? ( sys-libs/db:5[cxx] ) )
+		( sys-libs/db:4 tools? ( sys-libs/db:4[cxx] ) )
+	)
+"
+</codesample>
+
+</body>
+</subsection>
+
+<subsection>
+<title>The <c>:=</c> slot operator with multiple slots</title>
+<body>
+
+<p>
+The <c>:=</c> operator records the slot/sub-slot of the best matching
+installed version for the given atom (see PMS). If that atom can match
+newer slots than the versions you explicitly allowed elsewhere, it may bind
+to the wrong slot (and even pull it in during build).
+</p>
+
+<p>
+To prevent this, ensure the atom carrying <c>:=</c> cannot match newer slots
+than intended. One simple pattern is to cap it with an upper bound that
+excludes unwanted major versions while keeping <c>:=</c> on the same
+package:
+</p>
+
+<codesample lang="ebuild">
+# Good:
+DEPEND="
+	|| (
+		=sys-libs/db-5*
+		=sys-libs/db-4*
+	)
+	&lt;sys-libs/db-6:=
+"
+</codesample>
+
+<p>
+This forces the slot-operator binding to a version in the requested range at
+build time. Be cautious when combining multiple conditional USE sets with a
+slot operator; keeping the <c>:=</c> atom simple and separately constrained
+is usually clearer.
+</p>
+
+</body>
+</subsection>
+
+<subsection>
+<title>Understanding any-of dependencies</title>
+<body>
+
+<p>
+An any-of group (<c>|| ( ... )</c>) guarantees only that at least one
+immediate child element is satisfied for the relevant dependency class. It
+does not guarantee which element will be chosen, nor does it bind the choice
+made at build time to the choice used later at runtime. Order can be used as
+an implementation hint but is not a contract.
+</p>
+
+<codesample lang="ebuild">
+DEPEND="
+	|| (
+		dev-libs/A
+		dev-libs/B
+		dev-libs/C
+	)
+"
+</codesample>
+
+<p>
+If more than one alternative is installed, the package manager may use any
+of them. Replacing one alternative with another later still satisfies the
+dependency and should not be assumed to force rebuilds unless you have
+expressly tied the dependency via slots/sub-slots (outside of any-of) or via
+other mechanisms.
+</p>
+
+</body>
+</subsection>
+
+<subsection>
+<title>Do not use <c>:=</c> inside any-of groups</title>
+<body>
+
+<warning>
+Do not place <c>:=</c> atoms inside <c>|| ( ... )</c> groups. The semantics
+of the slot operator (binding to the slot/sub-slot of the installed match)
+conflict with the semantics of any-of (only one child needs to match and may
+change later). As a result, the requirements cannot be satisfied reliably and
+behaviour across package managers is undefined.
+</warning>
+
+<p>
+Instead, keep the any-of block free of slot operators and add a separate,
+well-constrained atom carrying <c>:=</c> if you need rebuild tracking.
+</p>
+
+<codesample lang="ebuild">
+# Bad:
+RDEPEND="
+	|| (
+		dev-libs/A:=
+		dev-libs/B:=
+	)
+"
+
+# Better:
+RDEPEND="
+	|| (
+		dev-libs/A
+		dev-libs/B
+	)
+	dev-libs/A:=
+"
```

Comment:
Yeah, I don't think this illustrates anything. In the end, B is completely redundant here, except that it technically may end up being installed via some misbehavior.